frontend/packages/console-app/src/components/cloud-shell/CloudShellExec.tsx

```diff
@@ -74,13 +76,33 @@ const CloudShellExec: React.FC<CloudShellExecProps> = ({
 
   const tick = useActivityTick(workspaceName, namespace);
 
-  const onData = React.useCallback(
-    (data: string): void => {
-      tick();
-      ws.current?.send(`0${Base64.encode(data)}`);
-    },
-    [tick],
-  );
+  React.useEffect(() => {
+    let tickOnInterval = setInterval(tick, TICK_INTERVAL);
+
+    // Function to handle ticking
+    const handleTickOnVisible = () => {
+      if (document.visibilityState === 'hidden' || !isActiveTab) {
+        clearInterval(tickOnInterval);
+        tickOnInterval = null;
+      }
+      if (document.visibilityState === 'visible' && isActiveTab) {
+        tickOnInterval = tickOnInterval || setInterval(tick, TICK_INTERVAL);
+      }
+    };
+
+    document.addEventListener('visibilitychange', handleTickOnVisible, false);
+    handleTickOnVisible();
+
+    return () => {
+      clearInterval(tickOnInterval);
+      tickOnInterval = null;
+      document.removeEventListener('visibilitychange', handleTickOnVisible);
+    };
+  }, [isActiveTab, tick]);
```

Comment:
will it make sense to use `requestAnimationFrame` here? can avoid listeners on `visibilitychange` and `setInterval`.

WDYT? 

cc @sahil143 @karthikjeeyar 

```suggestion
  React.useEffect(() => {
   let startTime;
   const handleTick = (timestamp) => {
   if ((!startTime || timestamp - startTime >= TICK_INTERVAL) && isActiveTab) {
     startTime = timestamp;
     tick();
    }
    requestAnimationFrame(handleTick);
   }
  
  requestAnimationFrame(handleTick);
  
  return () => {
    cancelAnimationFrame(handleTick);
  }
  }, [isActiveTab, tick]);
```

Reply:
Hi @invincibleJai , not able to unit test this out correctly.

Reply:
setTimeout is unit testable at jest. Although we are using a debounce but the requestAnimationFrame is triggered at around 60fps, do you think it is a good fit for the case here?